main.cpp

```diff
@@ -282,199 +299,187 @@ struct Bank
 
     void clip(float requiredClipStrength = 20.f);
-    bool canClip(float requiredClipStrength = 20.f); // returns whether these clippers' clip strength meets or exceeds the required clip strength
+    bool canClip(float requiredClipStrength = 20.f);
     void unfold();
     void fold();
 };
 
-/*
-Thing 7) Magnifying glass
-5 properties:
-    1) lens major radius (float)
-    2) lens light refraction angle (float)
-    3) lens magnification factor (float)
-    4) arm material (std::string)
-    5) glass type (std::string)
-3 things it can do:
-    1) magnify image
-    2) focus light
-    3) flip image
- */
+void NailClippers::clip(float requiredClipStrength)
+{
+    if ( canClip(requiredClipStrength) )
+    {
+        // clip the nail!
+    }
+}
+
+bool NailClippers::canClip(float requiredClipStrength)
+{
+    return ( requiredClipStrength < clipStrength ) ? true : false;
+}
+
+void NailClippers::unfold() { /* unfold impl */ }
+void NailClippers::fold() { /* fold impl */ }
 
 struct MagnifyingGlass
 {
-    // lens major radius
     float lensRadius = 1.f;
-    // lens light refraction angle
     float refractionAngle = 30.f;
-    // lens magnification factor
     float magFactor = 2.5f;
-    // arm material
     std::string armMaterial = "Stainless Steel";
-    // glass type
     std::string glassType = "Clear";
 
     void magnifyImage(float magnificationFactor = 2.f);
     void focusLight();
     void flipImage();
 };
 
-/*
-Thing 8) Hair comb
-5 properties:
-    1) length (float)
-    2) material (std::string)
-    3) number of teeth (int)
-    4) teeth pitch (float)
-    5) teeth length (float)
-3 things it can do:
-    1) comb hair
-    2) unfold
-    3) fold
- */
+void MagnifyingGlass::magnifyImage(float magnificationFactor)
+{
+    // magnify impl
+}
+
+void MagnifyingGlass::focusLight()
+{
+    // focus light impl
+}
+
+void MagnifyingGlass::flipImage()
+{
+    // flippy stuff
+}
 
 struct HairComb
 {
-    // length
     float length = 5.5f;
-    // material
     std::string material = "Aluminum";
-    // number of teeth
     int numTeeth = 30;
-    // teeth pitch
-    float teethPitch = .0125f;
-    // teeth length
     float teethLength = 1.125f;
 
     void combHair(int numStrokes = 1);
     void unfold();
     void fold();
-    double getTotalTeethLength(float numTeeth, float pitch);
+    double getTeethPitch(float numTeethInComb, float length);
 };
 
-/*
-Thing 9) Toothpick
-5 properties:
-    1) length (float)
-    2) material (std::string)
-    3) breaking strength (float)
-    4) tip angle (float)
-    5) number of times used (int)
-3 things it can do:
-    1) pick teeth
-    2) poke hole
-    3) snap
- */
+void HairComb::combHair(int numStrokes)
+{
+    // comb hair 'numStrokes' times!
+}
+
+void HairComb::unfold() { /* unfold impl */ }
+void HairComb::fold() { /* fold impl */ }
+
+double HairComb::getTeethPitch(float numTeethInComb, float lengthOfComb)
+{
+    return (double)lengthOfComb / (double)numTeethInComb;
+}
```

Comment:
There are 2 issues here:
- Casting has not been covered yet.
- you're using C-style casting, which is bad. 

Change your function argument types to `double` to eliminate the casting.
- [ ] revise